pkg/pipeline/encode/encode_prom.go

```diff
@@ -122,34 +133,116 @@ func (e *encodeProm) EncodeGauge(metric config.GenericMap) []interface{} {
 		for _, t := range gaugeInfo.tags {
 			entryLabels[t] = fmt.Sprintf("%v", metric[t])
 		}
-		entry := gaugeEntryInfo{
-			gaugeName:  e.prefix + gaugeName,
-			gaugeValue: valueFloat,
-			labels:     entryLabels,
+		entry := entryInfo{
+			eInfo: entrySignature{
+				Name:   e.prefix + gaugeName,
+				Labels: entryLabels,
+			},
+			value: valueFloat,
 		}
 		out = append(out, entry)
 		// push the metric to prometheus
-		// TODO - fix the types here
 		if gaugeInfo.promGauge != nil {
 			gaugeInfo.promGauge.With(entryLabels).Set(valueFloat)
 		}
-	}
 
+		cEntry := e.saveEntryInCache(entry, entryLabels)
+		cEntry.metricType = api.PromEncodeOperationName("Gauge")
+		cEntry.promGauge = gaugeInfo.promGauge
+	}
 	return out
 }
 
+func (e *encodeProm) saveEntryInCache(entry entryInfo, entryLabels map[string]string) *metricCacheEntry {
+	// save item in cache; use eInfo as key to the cache
+	var cEntry *metricCacheEntry
+	secs := time.Now().Unix()
+	eInfoBytes, _ := json.Marshal(&entry.eInfo)
```

Comment:
A bit expensive operation ... maybe there is some simpler way to get a "signature" ... maybe just use print of the tags and name ... this should take much less resources.

Also, there are "labels" and "tags" all over .. can we agree on a name for those and make sure we are consistent

Reply:
Changed `tags` to `labelNames`.  Changed Marshal to fmt.Sprintf, as suggested.